assignment/core/src/main/java/com/assignment/core/models/BreadcrumbModel.java

```diff
@@ -0,0 +1,72 @@
+package com.assignment.core.models;
+
+import com.day.cq.wcm.api.Page;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.models.annotations.*;
+import org.apache.sling.models.annotations.injectorspecific.*;
+
+import java.awt.desktop.ScreenSleepEvent;
+import java.util.*;
+
+@Model(adaptables = Resource.class, defaultInjectionStrategy = DefaultInjectionStrategy.OPTIONAL)
+public class BreadcrumbModel {
+
+    @SlingObject
+    private ResourceResolver resolver;
+
+    @Self
+    private Resource currentResource;
+
+    public List<PathSegment> getNavigationPath() {
+        List<PathSegment> segments = new LinkedList<>();
+
+        Resource pageResource = currentResource;
+
+        if (pageResource == null) {
+            return segments;
+        }
+
+        Resource currentPage = getCurrentPage(pageResource);
```

Comment:
Use PageManager API.

Reply:
in my new commit i have used page manager Apis